src/python/pants/option/parser.py

```diff
@@ -848,7 +854,11 @@ def check(val):
                     f"`{val}` is not an allowed value for option {dest} in {self._scope_str()}. Must be one of: {choices}"
                 )
 
-            if type_arg == file_option:
+            if type_arg == list:
+                if inspect.isclass(member_type) and issubclass(member_type, Enum):
+                    if len(merged_val) != len(set(merged_val)):
+                        raise ParseError(f"Duplicate enum values specified in list: {merged_val}")
```

Comment:
Good thinking!

Reply:
We support a `set` type... it would probably be better to allow duplicates in `list`, and have people use `set` if they don't want dupes?

Reply:
Do we? `type=set, member_type=MyEnum`? That will use `ListValueComponent` still?

Is there any use case where you can imagine the call site wanting to allow duplicate enum values?

Reply:
> Is there any use case where you can imagine the call site wanting to allow duplicate enum values?

Something imperative... "do x, then y, then x again".